### DISCUSSION TITLE
Determining .NET version on hosted servers

### DISCUSSION THREAD

**Discussion by budhoward** [dnnsoftware/Dnn.Platform#4644] [Q&A]
My hosting provider indicates. NET version 4.8.3761 is installed, but the DNN version info in the persona bar indicates I have .NET 4.6 (DNN version 9.3.2).

Is there another way to identify .NET version (without administrative access to server)?

**Comment by valadas** [comment]
Don't quote me on this, but I believe in the Persona Bar it indicates which .Net version is currently targeted by your DNN instance, In your case 4.6, when you upgrade, you will net .Net Framework 4.7.2, this is the minimal version required, so since you have 4.8 available you should be all good for both.

**Reply by 21Junipers** [reply]
I've run into the same issue as budhoward. Perhaps that is true - that hovering over DNN in the Persona Bar indicates the version "targeted" vs the version that the site is running under. If so, I'd respectfully suggest that needs to be changed as the other information shown on hover is: the current (DNN) version number , server name, and license key (if Evoq). So in only one case is the information reported not related to "current state".

For "DNN developers", a non-issue probably. New support people, or those with one or two, typically hosted DNN sites, attempting to follow the suggested upgrade path will likely go through a ticket cycle with their host provider.
![image](https://user-images.githubusercontent.com/9954947/116879016-ba83ea80-abe5-11eb-92a9-cdf892f28c4e.png)

**Reply by bdukes** [reply]
It looks like in 9.4.0 this logic was adjusted to support newer versions of the .NET Framework (#2712). It does appear to be the latest version available on the machine, not necessarily the version targeted by the DNN application. But the logic for finding that version is comparing known IDs against a list from Microsoft, so there wasn't a way to display the correct .NET version before information about that .NET version was known. Now that 4.8 is that last major version of .NET Framework, there shouldn't be a scenario where it gets out of sync again. Unfortunately, this doesn't help the scenario where you're upgrading from an older version of DNN, but I don't think there's much we can do there.,

**Comment by jeremy-farrance** [comment]
I think, I could be wrong, but somewhere around 9.1 to 9.4 Dnn got better at reporting the framework version. So if you are on an older version, you will see a more accurate version after upgrading. Also be aware the version it shows still might not be what you expect. I have multiple 9.09.01 sites and Dnn still just says 4.7. 
![image](https://user-images.githubusercontent.com/653301/116886825-2dde2a00-abef-11eb-96bf-e04bb9e0d2ec.png)

More details here. **How to: Determine which .NET Framework versions are installed**
https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed
 
It only takes a minute to use RegEdit to see the Release number and confirm what is actually installed. Or you can [do it in code](https://docs.microsoft.com/en-us/dotnet/framework/migration-guide/how-to-determine-which-versions-are-installed#query-the-registry-using-code).
